projects/eval-pqmon/src/common/iio_pqm.c

```diff
@@ -0,0 +1,1039 @@
+ * @param buf       - Buffer to be filled with requested data
+ * @param len 	    - Length of the received command buffer in bytes
+ * @param channel   - Command channel info
+ * @param attr_id   - attribute descriptor
+ * @return          - The length of the buffer in case of success, negative
+ * value otherwise
+ */
+int read_v_consel_attr(void *device, char *buf, uint32_t len,
+		       const struct iio_ch_info *channel, intptr_t attr_id)
+{
+	struct pqm_desc *desc;
+	if (!device)
+		return -ENODEV;
+	desc = device;
+	if (attr_id < PQM_DEVICE_ATTR_NUMBER) {
+		int available_id = desc->pqm_global_attr[attr_id];
+		strncpy(buf, pqm_v_consel_available[available_id], len);
+		return strlen(buf);
+	} else {
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Write the v_consel device attribute.
+ *
+ * @param device    - The iio device structure
+ * @param buf       - Buffer to be filled with requested data
+ * @param len 	    - Length of the received command buffer in bytes
+ * @param channel   - Command channel info
+ * @param attr_id   - attribute descriptor
+ * @return          - The length of the buffer in case of success, negative
+ * value otherwise
+ */
+int write_v_consel_attr(void *device, char *buf, uint32_t len,
+			const struct iio_ch_info *channel, intptr_t attr_id)
+{
+	struct pqm_desc *desc;
+	if (!device)
+		return -ENODEV;
+	desc = device;
+	int data_size = NO_OS_ARRAY_SIZE(pqm_v_consel_available);
+	for (int i = 0; i < data_size; i++) {
+		if (strcmp(buf, pqm_v_consel_available[i]) == 0) {
+			desc->pqm_global_attr[attr_id] = i;
+			return len;
+		}
+	}
+	return -EINVAL;
+}
+
+/**
+ * @brief Read the flicker_model device attribute.
+ *
+ * @param device    - The iio device structure
+ * @param buf       - Buffer to be filled with requested data
+ * @param len 	    - Length of the received command buffer in bytes
+ * @param channel   - Command channel info
+ * @param attr_id - attribute descriptor
+ * @return          - The length of the buffer in case of success, negative
+ * value otherwise
+ */
+int read_flicker_model_attr(void *device, char *buf, uint32_t len,
+			    const struct iio_ch_info *channel,
+			    intptr_t attr_id)
+{
+	struct pqm_desc *desc;
+	if (!device)
+		return -ENODEV;
+	desc = device;
+	if (attr_id < PQM_DEVICE_ATTR_NUMBER) {
+		int available_id = desc->pqm_global_attr[attr_id];
+		strncpy(buf, pqm_flicker_model_available[available_id], len);
+		return strlen(buf);
+	} else {
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Write the flicker_model device attribute.
+ *
+ * @param device    - The iio device structure
+ * @param buf       - Buffer to be filled with requested data
+ * @param len 	    - Length of the received command buffer in bytes
+ * @param channel   - Command channel info
+ * @param attr_id   - attribute descriptor
+ * @return          - The length of the buffer in case of success, negative
+ * value otherwise
+ */
+int write_flicker_model_attr(void *device, char *buf, uint32_t len,
```

Comment:
you don't need a function for each read and each write of each attribute, you can do this with a single read function and a single write function that manages all attributes and channels at once